storage/doc.go

```diff
@@ -48,6 +48,13 @@ an unauthenticated client with
 
     client, err := storage.NewClient(ctx, option.WithoutAuthentication())
 
+To use an emulator with this library, you can set the STORAGE_EMULATOR_HOST
```

Comment:
We should probably recommend that people use STORAGE_EMULATOR_HOST and give that as an example rather than the endpoint. I would give an example kind of like this:

```
// Set STORAGE_EMULATOR_HOST environment variable.
err := os.Setenv("STORAGE_EMULATOR_HOST", "localhost:9000")
if err != nil {
  // TODO: Handle error.
}

// Create client as usual.
client, err := storage.NewClient(ctx)
if err != nil {
  // TODO: Handle error.
}

// This request is now directed to http://localhost:9000/storage/v1/b
// instead of https://storage.googleapis.com/storage/v1/b
if err := client.Bucket("my-bucket").Create(ctx, projectID, nil); err != nil {
    // TODO: Handle error.
}
```